src/weapon/w_015.c

```diff
@@ -76,9 +166,314 @@ static void func_ptr_80170008(Entity* self) {
     }
 }
 
-INCLUDE_ASM("weapon/nonmatchings/w_015", func_ptr_8017000C);
+extern s16 D_6D000_8017A6FC[];
+extern s16 D_6D000_8017A71C[];
+
+void func_ptr_8017000C(Entity* self) {
+    Collider collider;
+    Primitive* prim;
+    s16 xOffset, yOffset;
+    s16 x, y;
+    Entity* e;
+
+    switch (self->step) {
+    case 0:
+        SetSpriteBank1(g_Animset);
+
+        self->animSet = ANIMSET_OVL(0x10);
+        self->unk5A = 0x64;
+        self->palette = PAL_DRA(0x110);
+
+        if (g_HandId) {
+            self->palette = PAL_DRA(0x128);
+            self->unk5A = 0x66;
+            self->animSet += 2;
+        }
+
+        self->facingLeft = PLAYER.facingLeft;
+        self->zPriority = PLAYER.zPriority - 2;
+        self->flags = FLAG_UNK_08000000 | FLAG_UNK_800000;
+
+        self->primIndex = g_api.AllocPrimitives(PRIM_TILE, 32);
+
+        if (self->primIndex == -1) {
+            self->flags &= ~FLAG_UNK_800000;
```

Comment:
I'm not sure why FLAG_UNK_800000 still exists, this should be FLAG_HAS_PRIMS

Reply:
I fixed this one and will submit another PR removing the `_UNK_` flag.